src/classes/CRLP_Rollup_SEL.cls

```diff
@@ -178,43 +234,69 @@ public class CRLP_Rollup_SEL {
-        for (Rollup__mdt rollup : cachedRollups) {
-            if (rollup.Active__c == true) {
-                if (rollup.Summary_Object__r.QualifiedApiName == objName) {
-                    String fldName = rollup.Summary_Field__r.QualifiedApiName;
-                    referencedFields.put(fldName, allFields.get(fldName.toLowerCase()));
-                }
-                if (rollup.Detail_Object__r.QualifiedApiName == objName && rollup.Detail_Field__c != null) {
-                    String fldName = rollup.Detail_Field__r.QualifiedApiName;
-                    referencedFields.put(fldName, allFields.get(fldName.toLowerCase()));
-                }
-                if (rollup.Amount_Object__r.QualifiedApiName == objName && rollup.Amount_Field__r != null) {
-                    String fldName = rollup.Amount_Field__r.QualifiedApiName;
-                    referencedFields.put(fldName, allFields.get(fldName.toLowerCase()));
-                }
-                if (rollup.Date_Object__r.QualifiedApiName == objName && rollup.Date_Field__r != null) {
-                    String fldName = rollup.Date_Field__r.QualifiedApiName;
-                    referencedFields.put(fldName, allFields.get(fldName.toLowerCase()));
-                }
-            }
+        // Collect SObjectFields in set in case there are duplicates
+        final Set<SObjectField> fields = new Set<SObjectField>();
+
+        // Always add the AccountId and Primary_Contact__c for the Opportunity object
+        if (objType == Schema.Opportunity.SObjectType) {
+            fields.add(Schema.Opportunity.AccountId);
+            fields.add(Schema.Opportunity.Primary_Contact__c);
         }
-        // Always include the CurrencyIsoCode field if it's available for the object
-        if (UserInfo.isMultiCurrencyOrganization() && allFields.containsKey('currencyisocode')) {
-            referencedFields.put('CurrencyIsoCode', allFields.get('currencyisocode'));
+
+        if (objType != null) {
+            final DescribeSObjectResult sObjectDescribe = UTIL_Describe.getObjectDescribe(objType);
+            final String sObjectName = sObjectDescribe.getName();
+            
+            // Note: DescribeSObjectResult.fields.getMap() is a case-insensitive map
+            final Map<String, SObjectField> fieldsByName = sObjectDescribe.fields.getMap();
+
+            // Always include the CurrencyIsoCode field if available for the object
+            if (UserInfo.isMultiCurrencyOrganization()
+                && fieldsByName.get('CurrencyIsoCode') != null
+            ) {
+                fields.add(fieldsByName.get('CurrencyIsoCode'));
+            }
+    
+            // Summary, Date, Amount, Detail Fields from cachedRollups
+            for (Rollup__mdt rollup : cachedRollups) {
+                if (rollup.Active__c == true) {
+                    // Summary Field
+                    if (rollup.Summary_Object__r.QualifiedApiName == sObjectName) {
```

Comment:
This is the one field that will always be required for a Rollup__mdt to exist, though I agree that it's worth verifying a not null condition just in case a customer admin manually clears the field from with Setup.